core/upgrade/src/main/java/org/opennms/upgrade/implementations/DiscoveryConfigurationLocationMigratorOffline.java

```diff
@@ -0,0 +1,120 @@
+package org.opennms.upgrade.implementations;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.nio.file.Paths;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.opennms.core.utils.ConfigFileConstants;
+import org.opennms.upgrade.api.AbstractOnmsUpgrade;
+import org.opennms.upgrade.api.OnmsUpgradeException;
+import org.w3c.dom.Document;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+public class DiscoveryConfigurationLocationMigratorOffline extends AbstractOnmsUpgrade {
+    private File m_configFile;
+    private File m_backupFile;
+
+    public final static String OLD_DEFAULT_LOCATION = "localhost";
+    public final static String NEW_DEFAULT_LOCATION = "Default";
+
+
+    public DiscoveryConfigurationLocationMigratorOffline() throws OnmsUpgradeException {
+        super();
+        m_configFile = Paths.get(ConfigFileConstants.getHome(), "etc", "discovery-configuration.xml").toFile();
+    }
+
+    @Override
+    public int getOrder() {
+        return 13;
+    }
+
+    @Override
+    public String getDescription() {
+        return "Changes the name for the default location from 'localhost' to 'Default'. See HZN-940.";
+    }
+
+    @Override
+    public void preExecute() throws OnmsUpgradeException {
+        log("Backing up discovery-configuration.xml\n");
+        m_backupFile = zipFile(m_configFile);
```

Comment:
Just for clarification: In our meeting yesterday I mentioned that `m_backupFile` is used in the `postExecute()` or `rollback()` phase and if `zipFile()` threw an exception we may run into a NPE. I verified today, that this can never occur, as if `preExecute()` fails, the execution is aborted.